aoc/pkg/advent/advent.go

```diff
@@ -0,0 +1,108 @@
+package advent
+
+import (
+	"aoc/pkg/structs"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func buildClient(urlObj *url.URL) (http.Client, error) {
+
+	sessionId := os.Getenv("AOC_SESSION_ID")
+	if sessionId == "" {
+		return http.Client{}, errors.New("no AOC_SESSION_ID provided. " +
+			"(It can be found by logging in through a browser and getting the session cookie.)",
+		)
+	}
+	jar, err := cookiejar.New(nil)
+	if err != nil {
+		return http.Client{}, err
+	}
+	client := http.Client{
+		Jar: jar,
+	}
+	sessionCookie := &http.Cookie{
+		Name:  "session",
+		Value: os.Getenv("AOC_SESSION_ID"),
+	}
+	client.Jar.SetCookies(urlObj, []*http.Cookie{sessionCookie})
+	return client, nil
+}
+
+func GetInputs(day int, year int) ([]structs.Group, error) {
+
+	urlObj, _ := url.Parse(fmt.Sprintf("https://adventofcode.com/%d/day/%d/input", year, day))
```

Comment:
Skipping error checks.

![](https://media.tenor.com/BIRrUUfHY9YAAAAd/key-and-peele-jordan-peele.gif)

Reply:
![image](https://user-images.githubusercontent.com/35393936/205518737-ac1db644-9287-40dc-9e82-09fa56f92db5.png)

Reply:
It should be fine as `day` and `year` are int. If they were string, it might contain characters invalid for an URL.

Reply:
For sure. I'd have to use some `UrlEncode` util in that case.